// Copyright (c) 2016-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the BSD-style license found in the
// LICENSE file in the root directory of this source tree. An additional grant
// of patent rights can be found in the PATENTS file in the same directory.
syntax = "proto3";

import "orc8r/protos/common.proto";

package magma.orc8r;

option go_package = "magma/orc8r/cloud/go/protos";

// --------------------------------------------------------------------------
// Directory Service
//
//
//
// --------------------------------------------------------------------------

enum TableID {
  IMSI_TO_HWID = 0;
  HWID_TO_HOSTNAME = 1;
}

message GetLocationRequest {
  string id = 1;
  TableID table = 2;
}

message DeleteLocationRequest {
  string id = 1;
  TableID table = 2;
}

message LocationRecord {
  // Object location. Client is responsible for location encoding and decoding
  string location = 1;
}

message UpdateDirectoryLocationRequest {
  string id = 1;
  LocationRecord record = 2;
  TableID table = 3;
}

service DirectoryService {

  // Get location of an object from the directory service
  // Throws UNKNOWN if object ID does not exist
  rpc GetLocation (GetLocationRequest) returns (LocationRecord) {};

  // Update the location record of an object in the directory service
  rpc UpdateLocation (UpdateDirectoryLocationRequest) returns (Void) {};

  // Delete location record of an object from the directory service
  // Throws UNKNOWN if object ID does not exist
  rpc DeleteLocation (DeleteLocationRequest) returns (Void) {};
}